features/admin.users.v1/components/AccountStatusFilterDropdown.tsx

```diff
@@ -0,0 +1,136 @@
+import ListItemText from "@oxygen-ui/react/ListItemText";
+import MenuItem from "@oxygen-ui/react/MenuItem";
+import Select from "@oxygen-ui/react/Select";
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import { DropdownChild } from "@wso2is/forms";
+import React, { FunctionComponent, ReactElement } from "react";
+import { useTranslation } from "react-i18next";
+import { USER_ACCOUNT_STATUS_FILTER_OPTIONS } from "../constants/user-management-constants";
+import "./AccountStatusFilterDropdown.scss";
+
+/**
+ * Props for the AccountStatusFilterDropdown component.
+ */
+interface AccountStatusFilterDropdownProps extends IdentifiableComponentInterface {
+    /**
+     * Array of selected account status filter keys.
+     */
+    selectedFilters: string[];
+    /**
+     * Callback function triggered when the selected filters change.
+     * @param selected - Array of selected filter keys.
+     */
+    onChange: (selected: string[]) => void;
+}
+
+/**
+ * The AccountStatusFilterDropdown component.
+ *
+ * @param props - Props injected into the component.
+ * @returns React Element.
+ */
+const AccountStatusFilterDropdown: FunctionComponent<AccountStatusFilterDropdownProps> = ({
+    selectedFilters,
+    onChange,
+    ["data-componentid"]: componentId = "account-status-filter-dropdown"
+}: AccountStatusFilterDropdownProps): ReactElement => {
+
+    const { t } = useTranslation();
+
+    const handleChange = (event: SelectChangeEvent<string[]>) => {
+        const value: string[] | string = event.target.value;
+
+        onChange(typeof value === "string" ? value.split(",") : value);
+    };
+
+    /**
+     * Renders the dropdown label with selection count badge.
+     * @returns The rendered component
+     */
+    const renderLabelWithCount = (): ReactElement => (
+        <Box
+            className="dropdown-label-container"
+            data-componentid={ `${componentId}-label` }
+        >
+            { t("users:advancedSearch.accountStatusFilter.label") }
+            { selectedFilters.length > 0 && (
+                <Badge
+                    className="status-count-badge"
+                    badgeContent={ selectedFilters.length }
+                    data-testid={ `${componentId}-label-badge` }
+                />
+            ) }
+        </Box>
+    );
+
+    return (
+        <Box
+            className="account-status-filter-dropdown"
+            data-componentid={ componentId }
+        >
+            <FormControl fullWidth>
+                <Select
+                    className="account-status-select-input"
+                    multiple
+                    value={ selectedFilters }
+                    onChange={ handleChange }
+                    renderValue={ renderLabelWithCount }
+                    data-testid={ `${componentId}-select-input` }
+                    displayEmpty
+                >
+                    { USER_ACCOUNT_STATUS_FILTER_OPTIONS
+                        .sort((
```

Comment:
remove the sorting , as it's hinders the user experience.

Reply:
https://github.com/wso2/identity-apps/pull/8171/commits/a35f3f308db99f0f624552fce3446dbdefa8b471